velox/dwio/common/ColumnSelector.cpp

```diff
@@ -248,7 +248,6 @@ void ColumnSelector::setReadAll() {
 }
 
 const ColumnFilter& ColumnSelector::getProjection() const {
-  DWIO_ENSURE(!filter_.empty(), "need a valid filter");
```

Comment:
It is possible to have an empty projection as in `select count(*)` case. Also, this check is present only here. Other uses of `filter_` like in `ColumnSelector::buildSelectedReordered()` do not have this.